util.cc

```diff
@@ -838,82 +778,57 @@ QString strip_html(const QString& utfstring)
   doc.setHtml(utfstring);
   return doc.toPlainText().simplified();
 #else
-  char* out;
-  char* instr;
-  char tag[8];
-  unsigned short int taglen = 0;
-
-  char* incopy = instr = xstrdup(utfstring);
-  /*
-   * We only shorten, so just dupe the input buf for space.
-   */
-  char* outstring = out = xstrdup(utfstring);
-
-  tag[0] = 0;
-  while (*instr) {
-    if ((*instr == '<') || (*instr == '&')) {
-      tag[0] = *instr;
-      taglen = 0;
+  static const QRegularExpression re("(?:<(?<tag>.*?)>)|(?:&(?<entity>.*?);)|(?<other>[^<&]+)|(?<fragment>.+)",
+                                     QRegularExpression::DotMatchesEverythingOption);
+  assert(re.isValid());
+  static const QRegularExpression newlinespace_re("\\n\\s*");
+  assert(newlinespace_re.isValid());
+  QString out;
+
+  QRegularExpressionMatchIterator it = re.globalMatch(utfstring);
+  while (it.hasNext()) {
+    auto match = it.next();
+    //qDebug() << match.capturedTexts();
+    // TODO: Qt >= 6.3 use match.hasCaptured(...) instead of !match.captured(...).isNull()
+    if (!match.captured(u"tag").isNull()) {
+      QString tag = match.captured(u"tag");
+      //qDebug() << "tag match:" << tag;
+      if (tag.startsWith("p", Qt::CaseInsensitive)) {
```

Comment:
Do we want to only match P tags and not PRE and PARAM tags?

if ((tag == "p") || tag.startsWith("p ", Qt::CaseInsensitive)) {

Reply:
if ((tag.compare("p", Qt::CaseInsensitive) == 0) || tag.startsWith("p ", Qt::CaseInsensitive)) {

Reply:
the full tag name is now captured and compare for equality. https://github.com/GPSBabel/gpsbabel/pull/1341/commits/06b8027d33dc6b58fcc71e384c909fe90dbcd068